pkg/client/ghcr/ghcr.go

```diff
@@ -47,42 +29,47 @@ func (c *Client) Tags(ctx context.Context, host, owner, repo string) ([]api.Imag
 	if err != nil {
 		panic(err)
 	}
-	client := github.NewClient(ghRateLimiter).WithAuthToken(c.Token)
+	client := github.NewClient(ghRateLimiter).WithAuthToken(opts.Token)
 
-	if repoExist(client, owner, repo) {
-		tags, err = getTagsFromRelease(client, owner, repo)
-		if err != nil {
-			return nil, err
-		}
-	} else {
-
-		tags, err = getTagsFromOrgPackage(client, owner, repo)
-		if err != nil {
-			return nil, err
-		}
+	return &Client{
+		client: client,
 	}
-	return tags, nil
 }
 
-func repoExist(client *github.Client, owner string, repo string) bool {
-	_, _, err := client.Repositories.Get(context.TODO(), owner, repo)
-	return err == nil
+func (c *Client) Name() string {
+	return "ghcr"
 }
 
-func getTagsFromOrgPackage(client *github.Client, owner string, repo string) ([]api.ImageTag, error) {
-	var tags []api.ImageTag
-	packageType := "container"
-	packageState := "active"
+func (c *Client) Tags(ctx context.Context, host, owner, repo string) ([]api.ImageTag, error) {
+	// Choose the correct list packages function based on whether the owner
+	// is a user or an organization
+	getAllVersions := c.client.Organizations.PackageGetAllVersions
+	user, _, err := c.client.Users.Get(ctx, owner)
```

Comment:
GitHub has a very aggressive rate limit - we should look to try and cache this somewhere/where possible as this could cause image discovery to be significantly delayed or fail.

Reply:
Addressed in efb6cc44b9ed2b5ee0fe173f4873e8aa72a2a826.